Issues with Blender 3.6 (A mesh with name Cube already exists.)

```
Python: Traceback (most recent call last):
  File "/var/folders/xg/tljx53lx3lx9hmdt29t03nf40000gn/T/1692334237.blend/Text", line 6, in <module>
  File "/Users/techdragon/Library/Application Support/Blender/3.6/scripts/addons/CodeToCADBlenderAddon/BlenderProvider.py", line 590, in createCube
    return self._createPrimitive("cube", "{},{},{}".format(width, length, height), **(keywordArguments or {}))
  File "/Users/techdragon/Library/Application Support/Blender/3.6/scripts/addons/CodeToCADBlenderAddon/BlenderProvider.py", line 576, in _createPrimitive
    BlenderActions.addPrimitive(
  File "/Users/techdragon/Library/Application Support/Blender/3.6/scripts/addons/CodeToCADBlenderAddon/BlenderActions.py", line 297, in addPrimitive
    assert blenderMesh == None, f"A mesh with name {primitiveName} already exists."
AssertionError: A mesh with name Cube already exists.
```

This mesh named Cube error appears for most of the examples when run in the Blender 3.6 LTS

@techdragon  I am adding a more helpful error message for the Default Cube situation. Does this look good?
```
"/Users/shehab/Desktop/projects/CodeToCAD/providers/blender/BlenderActions.py\", line 260, in blenderPrimitiveFunction\n    assert blenderMesh == None, f\"A mesh with name {primitiveName} already exists. {orphanMeshMessage}\"
AssertionError: A mesh with name Cube already exists. Your mesh is an orphan, please delete it.
If you are starting with the Default Cube, please remove it using Delete Hierarchy to delete both the object and the mesh, then try again.
```

--- a/providers/blender/BlenderActions.py
+++ b/providers/blender/BlenderActions.py
@@ -583,9 +583,13 @@ def createCollection(
     sceneName="Scene"
 ):
 
-    assert \
-        getCollection(name, sceneName) == None, \
-        f"Collection {name} already exists"
+    try:
+        existingCollection = getCollection(name, sceneName)
+        assert \
+            existingCollection == None, \
+            f"Collection {name} already exists"
+    except:
+        pass
 
     assert \
         sceneName in bpy.data.scenes, f"Scene {sceneName} does not exist"  # type: ignore

--- a/providers/blender/BlenderActions.py
+++ b/providers/blender/BlenderActions.py
@@ -241,6 +241,24 @@ def blenderPrimitiveFunction(
     **kwargs
 ):
 
+    primitiveName = primitive.defaultNameInBlender()
+
+    # Make sure an object or mesh with the same name don't already exist.
+    blenderObject = bpy.data.objects.get(primitiveName)
+    blenderMesh = bpy.data.meshes.get(primitiveName)
+
+    assert blenderObject == None, f"An object with name {primitiveName} already exists."
+
+    orphanMeshMessage = ""
+    if blenderMesh != None and blenderMesh.users == 0:
+        orphanMeshMessage += " Your mesh is an orphan, please delete it."
+
+        # issue-182, add a warning for the Default Cube:
+        if primitiveName == "Cube":
+            orphanMeshMessage += "If you are starting with the Default Cube, please remove both the object and the mesh using Delete Hierarchy (not just Delete), then try again."
+
+    assert blenderMesh == None, f"A mesh with name {primitiveName} already exists. {orphanMeshMessage}"
+
     if primitive == BlenderDefinitions.BlenderObjectPrimitiveTypes.cube:
         return bpy.ops.mesh.primitive_cube_add(size=1, scale=[dimension.value for dimension in dimensions[:3]], **kwargs)
 
@@ -286,15 +304,6 @@ def addPrimitive(
 ):
 
     assert primitiveType is not None, f"Primitive type is required."
-
-    primitiveName = primitiveType.defaultNameInBlender()
-
-    # Make sure an object or mesh with the same name don't already exist:
-    blenderObject = bpy.data.objects.get(primitiveName)
-    blenderMesh = bpy.data.meshes.get(primitiveName)
-
-    assert blenderObject == None, f"An object with name {primitiveName} already exists."
-    assert blenderMesh == None, f"A mesh with name {primitiveName} already exists."
 
     # Convert the dimensions:
     dimensionsList: list[Utilities.Dimension] = Utilities.getDimensionListFromStringList(

